### ISSUE TITLE
ec2: group_id doesn't seem to accept a list of security groups as documented

### ISSUE THREAD

**Issue by ralfbergs** [ansible/ansible#19119]
_From @ralfbergs on August 17, 2016 13:59_

##### ISSUE TYPE

- Bug Report
##### COMPONENT NAME

ec2
##### ANSIBLE VERSION

```
ansible 2.1.1.0
  config file =
  configured module search path = Default w/o overrides
```
##### SUMMARY

Ansible doesn't accept a list of security groups when launching an instance, while it's [documented](https://docs.ansible.com/ansible/ec2_module.html) that it does: "security group id (or list of ids) to use with the instance"
##### STEPS TO REPRODUCE

Execute the following task:

```
    - name: "Launch proxy instance: {{ owner }}_i_{{ env }}_dmz_2"
      ec2:
        region: "{{ region }}"
        image: "{{ ami_id }}"
        count_tag:
          Name: "{{ owner }}_i_{{ env }}_dmz_2"
        exact_count: 1
        #wait: yes
        instance_type: "t2.micro"
        key_name: "{{ ssh_key_name}}"
        # TODO
        group_id:
          - "{{ preprod_sg_ssh.group_id }}"
          - "{{ preprod_sg_proxy.group_id }}"
        vpc_subnet_id: "{{ preprod_subnet_dmz_2 }}"
        zone: "{{ az2 }}"
        instance_tags:
          Name: "{{ owner }}_i_{{ env }}_dmz_2"
          Env: "{{ owner }}_{{ env }}"
          Tier: "{{ owner }}_{{ env }}_dmz"
      register: preprod_i_dmz_2 # preprod_i_dmz_2.tagged_instances[0].id
```

##### EXPECTED RESULTS

Expected that the two SGs specified would be assigned to the instance.
##### ACTUAL RESULTS

None of the two SGs were assigned to the instance. The instance had the default SG assigned.

_Copied from original issue: ansible/ansible-modules-core#4451_

**Comment by ralfbergs** [comment]
_From @ansibot on August 17, 2016 14:1_



@ansible ping, this issue is waiting for your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ralfbergs** [comment]
_From @ansibot on September 8, 2016 18:58_

@ansible, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ralfbergs** [comment]
_From @ansibot on October 5, 2016 15:18_

@ansible, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ralfbergs** [comment]
_From @ansibot on October 21, 2016 8:41_

@ansible, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ralfbergs** [comment]
_From @ansibot on November 6, 2016 9:40_

@ansible, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ralfbergs** [comment]
_From @ansibot on November 22, 2016 9:49_

@ansible, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ralfbergs** [comment]
_From @ansibot on December 7, 2016 18:3_

This repository has been locked. All new issues and pull requests should be filed in https://github.com/ansible/ansible

Please read through the <a href="http://docs.ansible.com/ansible/dev_guide/repomerge.html#move-issues-and-prs-to-new-repo">repomerge</a> page in the dev guide. The guide contains links to tools which automatically move your issue or pull request to the ansible/ansible repo.

**Comment by s-hertel** [comment]
I wasn't able to reproduce your particular problem.
Some questions:
1. What version of boto are you using?
2. Does this problem persist if you remove `vpc_subnet_id: "{{ preprod_subnet_dmz_2 }}"` altogether from your task? 
3. Do you get the same error if you use the group names with the `group` option instead of `group_id`?

This is the playbook I tried to reproduce with:
```
---
- hosts: localhost
  connection: local
  tasks:
    - name: create ec2 instance with region us-east-1
      ec2:
        region: "{{ region }}"
        key_name: "{{ key }}"
        instance_type: t2.micro
        image: ami-0b33d91d
        exact_count: 1
        count_tag:
          Name: "{{ tag }}"
        vpc_subnet_id: "{{ vpc_subnet }}"
        zone: "{{ zone }}"
        group_id:
          - "{{ id1 }}"
          - "{{ id2 }}"
```
ansible version: `ansible 2.1.1.0 (test 780c363482) last updated 2017/03/08 11:53:05 (GMT -400)`

needs_info

**Comment by ansibot** [comment]
@ralfbergs This issue is waiting for your response. Please respond or the issue will be closed.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@ralfbergs You have not responded to this issue so we will assume it no longer affects you. If you are still interested in this, please create a new issue.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ralfbergs** [comment]
Apologies, but I currently don't have the chance to work on this again.

Thanks for your efforts.

If I have the chance to revisit this, I will update this issue accordingly.
